### DISCUSSION TITLE
error

### DISCUSSION THREAD

**Discussion by HuskyXaHusky** [terser/html-minifier-terser#134] [Q&A] [answered]
Hello!
Faced a problem. I fed pages to various minifiers through the terminal. In them and in this, if the minifier meets in the html code:

<i class"pring"="">

It cannot parse and throws an error. I found this everywhere on the site with instructions for preparations.

PS For some of my needs, I'm putting together some instructions. I had to write a program that cuts out everything superfluous so that it is more convenient to work with the material, and also so that the minifier does not choke. But it would be great if the minifier could easily eat such a "code".

**Comment by DanielRuf** [comment, marked as answer]
Hi @HuskyXaHusky,

thanks for reporting this.

This tool can only parse valid HTML.
`<i class"pring"="">` is not valid HTML code, but `<i class="pring">` is.

**Reply by HuskyXaHusky** [reply]
Thanks @DanielRuf for the quick response. It is clear now why he did not process the code.
 However, I'm surprised that minifiers don't ignore invalid code.

**Reply by DanielRuf** [reply]
To minify they have to parse the code. To parse it they need valid syntax to build an AST (abstract syntax tree) from it. Otherwise the code can not be understood.

If you want to ignore specific parts from being minified and parsed, there is a way using HTML comments: https://github.com/terser/html-minifier-terser#ignoring-chunks-of-markup
